docker/Dockerfile-pytorch

```diff
@@ -1,21 +1,28 @@
 ARG base_image_tag
-FROM twosixarmory/base:${base_image_tag} as armory-base
 ARG armory_version
 ARG armory_build_type
 
-FROM armory-base AS armory-local
-RUN echo "Installing ART"
-RUN /opt/conda/bin/pip install --no-cache-dir \
-    adversarial-robustness-toolbox==1.10.3
 
-RUN echo "Building Armory from local source"
-WORKDIR /armory-repo/
-COPY setup.py LICENSE MANIFEST.in README.md requirements.txt test-requirements.txt armory-base-requirements.txt /armory-repo/
-COPY pytest.ini /armory-repo/
-COPY ./tests /armory-repo/tests/
-COPY ./scenario_configs /armory-repo/scenario_configs/
-COPY ./armory /armory-repo/armory/
-RUN SETUPTOOLS_SCM_PRETEND_VERSION=${armory_version} /opt/conda/bin/pip install --use-feature=in-tree-build --no-cache-dir .
-RUN armory configure --use-default
+FROM twosixarmory/base:${base_image_tag} AS armory-local
+
+COPY dist /tmp
+
+WORKDIR /armory-repo
+
+RUN echo "Installing ART"                                               && \
+    pip install --no-cache-dir adversarial-robustness-toolbox==1.10.3   && \
+    echo "Building Armory from local source"                            && \
+    echo "Updating Base Image..."                                       && \
+      python -m pip install --upgrade pip                               && \
+    echo "Unpacking sdist into `/armory-repo`..."                       && \
+      tar -zxvf /tmp/*.tar.gz                                              \
```

Comment:
Where does this tar file come from? Is it set up somewhere else?

Reply:
The tar file(`sdist`) is the output from the build process. This happens in [`build.py`](https://github.com/christopherwoodall/armory/blob/develop/docker/build.py#L80) using `hatch`, but could easily switch to `python -m build`.

In saying that, I've modified the above approach. The [new approach](https://github.com/christopherwoodall/armory/actions/runs/3138111637) adds an exemption for the `.git` directory in [`.dockerignore`](https://github.com/christopherwoodall/armory/blob/build-system/.dockerignore), then uses a single [`COPY` command](https://github.com/christopherwoodall/armory/blob/build-system/docker/Dockerfile-pytorch#L10) to copy the necessary metadata for a local(in container) install.